pkg/artifactory/resource_artifactory_virtual_repository_test.go

```diff
@@ -138,6 +138,11 @@ func testAccCheckVirtualRepositoryDestroy(id string) func(*terraform.State) erro
 		}
 
 		repo, resp, err := client.V1.Repositories.GetVirtual(context.Background(), rs.Primary.ID)
+
+		if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.